src/client/api/event.c

```diff
@@ -1028,6 +1028,10 @@ daos_event_init(struct daos_event *ev, daos_handle_t eqh,
 		evx->evx_sched = &eqx->eqx_sched;
 		daos_eq_putref(eqx);
 	} else {
+		if (daos_sched_g.ds_udata == NULL) {
+			D_ERROR("Event queue is not initialized\n");
```

Comment:
At a high level, the segmentation fault is caused due to the DAOS client library is not initialized. In other words, `daos_init()` must be called before calling `daos_obj_open().`
Having said that, I suggest changing the message to something like 
`D_ERROR("The DAOS client library is not initialized"DF_RC"\n", DP_RC(-DER_EQ_UNINIT));`

Reply:
Done. I agree, this error message is more descriptive.